### PR TITLE
[Twig][Bridge][TranslationDefaultDomain] add support of named arguments.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #10859 |
| Tests pass? | yes |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php [hunk 1]
```diff
@@ -103,4 +109,16 @@ public function getPriority()
...
         return -10;
     }
+
+    /**
+     * @return bool
+     */
+    private function isNamedArguments($arguments)
+    {
+        foreach ($arguments as $name => $node) {
+            return !is_int($name);
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php]: This looks wrong to me as you always return during the first iteration.

**Comment by aitboudad** [comment]
done

**Comment by fabpot** [comment]
@aitboudad Can you add some functional tests as well?

**Comment by aitboudad** [comment]
functional tests added :)

**Comment by aitboudad** [comment]
case of mix positional and named arguments is done now

File: src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php [hunk 2]
```diff
@@ -71,6 +71,24 @@ public function testTransChoiceComplexBody()
         $output = $this->getTemplate("{% transchoice count %}\n{{ 1 + 2 }}{% endtranschoice %}")->render();
     }
 
+    /**
+     * @expectedException        \Twig_Error_Syntax
+     * @expectedExceptionMessage Positional arguments cannot be used after named arguments for filter "trans" in "index".
+     */
+    public function testTransPositionalArgumentsAfterNamedArguments()
+    {
+        $output = $this->getTemplate('{{- "foo"|trans(arguments = {}, "custom") }}')->render();
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php]: this test has nothing to do here. You are not testing the TranslationExtension here. You are testing Twig itself (where it is already tested).

File: src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php [hunk 3]
```diff
@@ -71,6 +71,24 @@ public function testTransChoiceComplexBody()
...
+    }
+
+    /**
+     * @expectedException        \Twig_Error_Syntax
+     * @expectedExceptionMessage Positional arguments cannot be used after named arguments for filter "transchoice" in "index".
+     */
+    public function testTransChoicePositionalArgumentsAfterNamedArguments()
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php]: same issue here

File: src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php [hunk 4]
```diff
@@ -148,6 +166,39 @@ public function testDefaultTranslationDomain()
...
+                {%- extends "base" %}
+
+                {%- trans_default_domain "foo" %}
+
+                {%- block content %}
+                    {{- "foo"|trans(arguments = {}, domain = "custom") }}
+                    {{- "foo"|transchoice(count = 1) }}
+                    {{- "foo"|transchoice(count = 1, arguments = {}, domain = "custom") }}
+                    {{- "foo"|trans({}, domain = "custom") }}
+                    {{- "foo"|transchoice(1, arguments = {}, domain = "custom") }}
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php]: you have at least 1 missing case here (and this case is broken according to my code review): passing the domain as a positional argument while using a named argument later in the argument list (for the locale for instance)

File: src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php [hunk 5]
```diff
@@ -148,6 +166,39 @@ public function testDefaultTranslationDomain()
         $this->assertEquals('foo (foo)foo (custom)foo (foo)foo (custom)foo (foo)foo (custom)', trim($template->render(array())));
     }
 
+    public function testDefaultTranslationDomainWithNamedArguments()
+    {
+        $templates = array(
+            'index' => '
+                {%- extends "base" %}
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php]: you could get rid of the inheritance in this test

File: src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php (hunk 4 above)
Comment (aitboudad) [inline src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php]: :+1:  fixed

**Comment by aitboudad** [comment]
I guess that all cases were tested :)

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request closed without merging**